classes/Pods.php

```diff
@@ -1103,51 +1036,111 @@ public function field( $name, $single = null, $raw = false ) {
 
 					$object_field_found = true;
 				}
-			} elseif ( 0 === strpos( $params->name, 'image_attachment.' ) ) {
-				$size = 'thumbnail';
-
-				$image_id = 0;
-
-				$field_names = explode( '.', $params->name );
+			}
 
-				if ( isset( $field_names[1] ) ) {
-					$image_id = $field_names[1];
-				}
+			if ( ! $object_field_found ) {
+				foreach ( $image_fields as $image_field ) {
+					if ( isset( $this->fields[ $image_field ] ) ) {
+						// Skip any names that are registered fields within this Pod.
+						continue;
+					}
+					if (
+						$image_field === $params->name ||
+						0 === strpos( $params->name, $image_field . '.' )
+					) {
+						// Is it a URL request?
+						$url = '_url' === substr( $image_field, -4 );
+						if ( $url ) {
+							$image_field = substr( $image_field, 0, -4 );
+						}
 
-				if ( isset( $field_names[2] ) ) {
-					$size = $field_names[2];
-				}
+						// Results in an empty array if no traversal names are passed.
+						$traverse_names = explode( '.', $params->name );
+						array_shift( $traverse_names );
+
+						$attachment_id = 0;
+						switch ( $image_field ) {
+							case 'post_thumbnail':
+								$attachment_id = get_post_thumbnail_id();
+								break;
+							case 'image_attachment':
+								if ( isset ( $traverse_names[0] ) ) {
```

Comment:
* :no_entry_sign: isset() must only contain variables (PHP 5.6 compatibility)
* :no_entry_sign: Space before opening parenthesis of function call prohibited

<sub>hash: 6c47d75ec661b98dd39761f43d050746</sub>